Loop/View Models/BolusEntryViewModel.swift

```diff
@@ -517,8 +512,25 @@ final class BolusEntryViewModel: ObservableObject {
             completion()
             return
         }
-        pumpManagerRefreshCompletion = completion
-        pumpManager.assertCurrentPumpData()
+        
+        DispatchQueue.main.async {
+            // v-- This needs to happen on the main queue
+            self.isRefreshingPump = true
+            let wrappedCompletion: () -> Void = { [weak self] in
+                self?.dataManager.loopManager.getLoopState { [weak self] _, _ in
+                    // v-- This needs to happen in LoopDataManager's dataQueue
+                    completion()
```

Comment:
This dispatching is pretty crazy. I do see why it's here; getting status from LoopDataManager returns an object that accesses shared data in LDM. I think at some point additional usages of getLoopState were added beyond the scope of what it was designed for. This might require some bigger thinking in terms of how loop state is shared for display, and how these "temporary" forecasts are generated.  I think they are two different use cases that should probably be supported by different calls.

Reply:
You ain't kidding :)  ... that's why I added to comments. I admit it took me a while to figure out why I was getting dispatch queue assertions until I saw other code using `getLoopState` to hop on the right queue (`dataQueue`) before proceeding.  In other words: this isn't the only place that does this.
It may be the case that my refactoring `BolusEntryViewModel` to be more unit testable might surface that "bigger thinking" you refer to.  Or...did you want to file a separate Jira ticket to cover it?